src/App.tsx

```diff
@@ -1,61 +1,127 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types';
+import { TodoList } from './components/TodoList';
+
+function getUserById(idOfUser: number) {
+  return usersFromServer.find(user => user.id === idOfUser) || null;
+}
+
+function getMaxTodoId(list: Todo[]) {
+  const listOfId = list.map(item => item.id);
+
+  return Math.max(...listOfId);
+}
+
+const initialList = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(initialList);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [hasUserError, setHasUserError] = useState(false);
+  const [userId, setUserId] = useState(0);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title.trim());
+    setHasUserError(!userId);
+
+    if (!title.trim() || !userId) {
+      // eslint-disable-next-line no-useless-return
+      return;
+    }
+
+    setTodos(currentList => [...currentList, {
+      id: getMaxTodoId(currentList) + 1,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId) || null,
```

Comment:
You can create func that return you ready todo
Then you can move it outside component file